src/AccountDeleter/AccountDeleteMessageHandler.cs

```diff
@@ -0,0 +1,122 @@
+
+                var username = command.Username;
+                var user = _userService.FindByUsername(username);
+                if (user == null)
+                {
+                    throw new UserNotFoundException();
+                }
+
+                if (_accountDeleteConfigurationAccessor.Value.RespectEmailContactSetting && !user.EmailAllowed)
+                {
+                    throw new EmailContactNotAllowedException();
+                }
+
+                var recipientEmail = user.EmailAddress;
+                var deleteSuccess = await _accountManager.DeleteAccount(user);
+                _telemetryService.TrackDeleteResult(source, deleteSuccess);
+
+                var baseEmailBuilder = _emailBuilderFactory.GetEmailBuilder(source, deleteSuccess);
+                if (baseEmailBuilder != null)
+                {
+                    var toEmail = new List<MailAddress>();
+
+                    var configuration = _accountDeleteConfigurationAccessor.Value;
+                    var senderAddress = configuration.EmailConfiguration.GalleryOwner;
+                    var ccEmail = new List<MailAddress>();
+                    // toEmail.Add(new MailAddress(recipientEmail)); // Temporarily disable sending to end user while we are in phase 1.
+                    toEmail.Add(new MailAddress(senderAddress)); // Remove this when we switch to phase 2.
+                    ccEmail.Add(new MailAddress(senderAddress));
+
+                    var recipients = new EmailRecipients(toEmail, ccEmail);
+                    var emailBuilder = new DisposableEmailBuilder(baseEmailBuilder, recipients, username);
+                    await _messenger.SendMessageAsync(emailBuilder);
+                    _telemetryService.TrackEmailSent(source, user.EmailAllowed);
+                }
+            }
+            catch (UnknownSourceException)
+            {
+                // Should definitely log if source isn't expected. Should we even send mail? or log and alert?
+                // Log unknown source and fail.
+                _logger.LogError("Unknown message source detected: {Source}.", command.Source);
+                _telemetryService.TrackUnknownSource(source);
+                messageProcessed = false;
+            }
+            catch (EmailContactNotAllowedException)
+            {
+                // Should we not send? or should we ignore the setting.
+                _logger.LogWarning("User did not allow Email Contact.");
+                _telemetryService.TrackEmailBlocked(source);
+            }
+            catch (UserNotFoundException)
+            {
+                _logger.LogWarning("User was not found. They may have already been deleted.");
+                _telemetryService.TrackUserNotFound(source);
+                messageProcessed = true;
+            }
+            catch (Exception e)
+            {
+                _logger.LogError("An unknown exception occured: {ExceptionMessage}", e.Message);
```

Comment:
Prefer `e.ToString()` over `e.Message` as the real error may be in an inner exception. In any case, I would use the logging framework:

```suggestion
                _logger.LogError(0, e, "An unknown exception occurred");
```

Reply:
Ah yes this is what should be used here. I think i did it somewhere else the right way.